tooltester/src/main/java/io/dockstore/tooltester/S3Client.java

```diff
@@ -0,0 +1,116 @@
+package io.dockstore.tooltester;
+
+import java.io.ByteArrayInputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.ws.rs.core.MediaType;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+
+/**
+ * @author gluu
+ * @since 18/04/19
+ */
+public class S3Client {
+    private static final int MAX_TOOL_ID_STRING_SEGMENTS = 5;
+    private static final int TOOL_ID_REPOSITORY_INDEX = 3;
+    private static final int TOOL_ID_TOOLNAME_INDEX = 4;
+    private String bucketName;
+    private AmazonS3 s3;
+
+    public S3Client() {
+        TooltesterConfig tooltesterConfig = new TooltesterConfig();
+        String s3Endpoint = tooltesterConfig.getS3Endpoint();
+        bucketName = tooltesterConfig.getS3Bucket();
+        AwsClientBuilder.EndpointConfiguration endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(s3Endpoint,
+                Regions.US_EAST_1.getName());
```

Comment:
The region could be configurable, although we could also just cross that bridge if we run into the need.